napi/src/threadsafe_function.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{check_status, ptr, sys, Env, Function, Result, Value};
+use std::os::raw::{c_char, c_void};
+
+// TODO Should we pub these enums? or pub the sys mod directly?
+pub use sys::napi_threadsafe_function_call_mode;
```

Comment:
@Brooooooklyn Do you think we should `pub sys` mod?

Reply:
I will publish `sys` as an independent crate `https://github.com/napi-rs/napi-rs/issues/55`

Reply:
BTW, We have already `pub mod sys;` in `lib.rs`

Reply:
And then I would like to stop `pub` these enums.